### PR TITLE
X1034 dep update

### PR DESCRIPTION
**Pull request by seenanair**
Dependency updates for some packages . The main ones are faker, msw, tailwind, typescript

- Changes associated with faker library  deprecation /change of many methods 
- Changes for TS - The generic type is not supported without constraint, so an extends keyword which clearly state what the user want a generic type is required

### CODE REVIEW

**Comment by khelwood** [comment]
Is this actually ready to review? It is marked as a draft

**Comment by seenanair** [comment]
> Is this actually ready to review? It is marked as a draft

Yes it is. Sorry, I will change the status

File: src/pages/Search.tsx [hunk 1]
```diff
@@ -170,7 +170,7 @@ function Search({ searchInfo, urlParamsString }: SearchProps) {
 
   const numDisplayed = viewAllRecords ? searchResult?.numDisplayed : memoUniqueBarcodeData.length;
   const numRecords = viewAllRecords ? searchResult?.numRecords : memoUniqueBarcodeData.length;
-
+  debugger;
```
Comment (khelwood) [inline src/pages/Search.tsx:173]: debugger

**Comment by seenanair** [comment]
> Happy to approve if the tests will actually pass

I will look into this issue

**Pull request merged**